pysph/tools/read_mesh.py

```diff
@@ -0,0 +1,92 @@
+'''
+Functions can be used to generate points to describe a given input
+mesh.
+
+Supported mesh formats: All file formats supported by meshio.
+(https://github.com/nschloe/meshio)
+'''
+
+import numpy as np
+import meshio
+from pysph.tools.mesh_tools import surface_points, surf_points_uniform
+
+
+class Mesh:
+    def __init__(self, file_name, file_type=None):
+        if file_type is None:
+            self.mesh = meshio.read(file_name)
+        else:
+            self.mesh = meshio.read(file_name, file_type)
+
+        self.cells = np.array([], dtype=int).reshape(0, 3)
+
+    def extract_connectivity_info(self):
+        cell_blocks = self.mesh.cells
+        for block in cell_blocks:
+            self.cells = np.concatenate((self.cells, block.data))
+
+        return self.cells
+
+    def extract_coordinates(self):
+        x, y, z = self.mesh.points.T
+        self.x, self.y, self.z = x, y, z
+
+        return x, y, z
+
+    def compute_normals(self):
+        n = self.cells.shape[0]
+        self.normals = np.zeros((n, 3))
+        points = self.mesh.points
+
+        idx = self.cells
+        pts = points[idx]
+        a = pts[:, 1] - pts[:, 0]
+        b = pts[:, 2] - pts[:, 0]
+
+        normals = np.cross(a, b)
+        mag = np.linalg.norm(normals, axis=1)
+        mag_arr = np.array([np.ones(3)*norm for norm in mag])
```

Comment:
This is not needed, you just need to do a small change and can use broadcasting to achieve this faster:
```
mag.shape = (n, 1)
self.normals = normals/mag
```